### PR TITLE
[API change] rename SpaceType to ActionType

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Describe the changes made in this PR.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added updated the changelog (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] I have updated the migration guide (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/ActionMasker.cs [hunk 1]
```diff
@@ -118,7 +118,7 @@ internal bool[] GetMask()
         void AssertMask()
         {
             // Action Masks can only be used in Discrete Control.
-            if (m_BrainParameters.vectorActionSpaceType != SpaceType.Discrete)
+            if (m_BrainParameters.vectorActionSpaceType != ActionType.Discrete)
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/ActionMasker.cs]: `ActionSpaceType`? That's what we're calling it in BrainParameters.

**Comment by chriselion** [comment]
Can you confirm that this doesn't break serialization for existing scenes?

**Comment by vincentpierre** [comment]
> Can you confirm that this doesn't break serialization for existing scenes?
Enums are serialized as int

File: com.unity.ml-agents/Runtime/ActionMasker.cs (hunk 1 above)
Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/ActionMasker.cs]: ActionSpaceType sounds weird to me. I think Action Type makes more sense to me as I am not sure where space comes from.
@andrewcoh

Comment (andrewcoh) [reply com.unity.ml-agents/Runtime/ActionMasker.cs]: People typically use space to refer to the set of actions an agent has i.e. continuous/discrete action spaces.  Also, gym organizes their action spaces in a directory 'spaces' https://github.com/openai/gym/tree/master/gym/spaces

You could argue that action space type refers to the nature of the space i.e. discrete/continuous but I'm not sure it's worth arguing

Comment (chriselion) [reply com.unity.ml-agents/Runtime/ActionMasker.cs]: ok, I guess you can parse `m_BrainParameters.vectorActionSpaceType` as (vector action) (space type), so "SpaceType" would be reasonable.

**Comment by chriselion** [comment]
What about SpaceTypeProto?

**Pull request closed without merging**